r/tests/testthat/test-dataset-csv.R

```diff
@@ -444,11 +444,35 @@ test_that("skip argument in open_dataset", {
 })
 
 test_that("error message if non-schema passed in as schema to open_dataset", {
-
   # passing in the schema function, not an actual schema
   expect_error(
     open_dataset(csv_dir, format = "csv", schema = schema),
     regexp = "`schema` must be an object of class 'Schema' not 'function'.",
     fixed = TRUE
   )
 })
+
+test_that("CSV reading/parsing/convert options can be passed in as lists", {
+  tf <- tempfile()
+  on.exit(unlink(tf))
+
+  writeLines('"x"\n"y"\nNA\nNA\n"NULL"\n\n"foo"\n', tf, )
```

Comment:
nit: does this trailing comma thing do anything?
```suggestion
  writeLines('"x"\n"y"\nNA\nNA\n"NULL"\n\n"foo"\n', tf)
```